prow/plugins/cherrypickunapproved/cherrypick-unapproved.go

```diff
@@ -0,0 +1,157 @@
+*/
+
+// Package cherrypickunapproved adds the `do-not-merge/cherry-pick-not-approved`
+// label to PRs against a release branch which do not have the
+// `cherrypick-approved` label.
+package cherrypickunapproved
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "cherrypick-unapproved"
+
+const (
+	cpUnapprovedLabel         = "do-not-merge/cherry-pick-not-approved"
+	cpApprovedLabel           = "cherrypick-approved"
+	labelUnapprovedFormat     = "This PR is not for the master branch but does not have the `%s` label. Adding the `%s` label.\n"
+	assignPatchReleaseManager = `
+	To approve the cherrypick, please assign the patch release manager for the ` + "`%s`" + ` branch by writing ` + "`/assign @username`" + ` in a comment when ready.
+The list of patch release managers for each release can be found [here](https://git.k8s.io/sig-release/release-managers.md).`
+)
+
+var (
+	labelUnapprovedBody = fmt.Sprintf(labelUnapprovedFormat, cpApprovedLabel, cpUnapprovedLabel)
+)
+
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequest, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	// Only the 'Config' and Description' fields are necessary because this
+	// plugin does not react to any commands.
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "Label `do-not-merge/cherry-pick-not-approved` to PRs against a release branch which do not have the `cherrypick-approved` label",
+		Config: map[string]string{
+			"": fmt.Sprintf(
+				"The cherrypick-unapproved plugin treats PRs against branch names satisfying the regular expression `%s` as cherrypick PRs.",
+				config.CherryPickUnapproved.BranchRegexp,
+			),
+		},
+	}
+	return pluginHelp, nil
+}
+
+type githubClient interface {
+	BotName() (string, error)
+	CreateComment(owner, repo string, number int, comment string) error
+	AddLabel(owner, repo string, number int, label string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+}
+
+type commentPruner interface {
+	PruneComments(shouldPrune func(github.IssueComment) bool)
+}
+
+func handlePullRequest(pc plugins.PluginClient, pr github.PullRequestEvent) error {
+	return handlePR(pc.GitHubClient, pc.Logger, &pr, pc.CommentPruner, pc.PluginConfig.CherryPickUnapproved.BranchRe)
+}
+
+func handlePR(gc githubClient, log *logrus.Entry, pr *github.PullRequestEvent, cp commentPruner, branchRe *regexp.Regexp) error {
+	// Only consider the events that indicate opening of the PR and
+	// when the cherrypick-approved label is added or removed
+	cpApprovedLabelUpdated := (pr.Action == github.PullRequestActionLabeled || pr.Action == github.PullRequestActionUnlabeled) && pr.Label.Name == cpApprovedLabel
```

Comment:
What if someone manually removes the `do-not-merge/cherry-pick-not-approved` label without adding the approved label?